addons/mail/models/mail_channel.py

```diff
@@ -651,7 +672,7 @@ def channel_get(self, partners_to, pin=True):
             # pin up the channel for the current partner
             if pin:
                 self.env['mail.channel.partner'].search([('partner_id', '=', self.env.user.partner_id.id), ('channel_id', '=', channel.id)]).write({'is_pinned': True})
-            channel._broadcast(self.env.user.partner_id.ids)
+            channel._broadcast(self.env.user.partner_id.ids, 'mail.pin_private_channel')
```

Comment:
It's more specific than "private" channel actually, this is specifically for a chat (dm).